StopwatchExerciseTests/StopwatchExerciseTests.swift

```diff
@@ -11,28 +11,59 @@ import Combine
 
 
 final class StopwatchExerciseTests: XCTestCase {
-    var timerPublisher: AnyPublisher<Publishers.Autoconnect<Timer.TimerPublisher>.Output, Publishers.Autoconnect<Timer.TimerPublisher>.Failure>!
-    var cancellable: AnyCancellable!
-    var watch: Watch!
     
-    override func setUp() {
-        timerPublisher = Timer.publish(every: 1.0, on: .main, in: .common).autoconnect().eraseToAnyPublisher()
-        cancellable = .none
-        watch = Watch(timerPublisher: timerPublisher)
+    func test_init_startWithCorrectValue() {
+        let (sut, _) = makeSUT()
+        
+        XCTAssertEqual(sut.remainingTime, 1500)
+        XCTAssertEqual(sut.isStarted, false)
+    }
+    
+    func test_startWatchAndWait30seconds() {
+        let (sut, publisher) = makeSUT()
+        
+        sut.start()
+        (1 ... 29).forEach { _ in publisher.sendValue() }
+        
+        XCTAssertEqual(sut.time, "24:30")
+        XCTAssertEqual(sut.isStarted, true)
+    }
+    
+    func test_pauseWatchAfter5seconds() {
+        let (sut, publisher) = makeSUT()
+        
+        sut.start()
+        (1 ... 4).forEach { _ in publisher.sendValue() }
+        sut.pause()
+        XCTAssertEqual(sut.time, "24:55")
+        XCTAssertEqual(sut.isStarted, false)
+    }
+    
+    func test_startWatchAndWait25MinutesToReset() {
+        let (sut, publisher) = makeSUT()
+        
+        sut.start()
+        // when remainingTime == 0 reset() is used
+        (1 ... 1499).forEach { _ in publisher.sendValue() }
```

Comment:
```suggestion
        (1 ... 2).forEach { _ in publisher.sendValue() }
```
Ok, ici pas besoin de faire une boucle qui tourne 1499 toit, crée un timer de 2 secondes et fait une boucle plus petite, le résultat sera le même et ton test sera plus optimisé.